tests/app_test.py

```diff
@@ -88,65 +33,169 @@ def test_find_valid_refresh_token(test_user, db_session):
 
 
-def test_token_endpoint_with_default_idp(client, test_user, db_session, auth_header):
-    logged_in_user_data = create_logged_in_user_data(test_user, db_session)
-    create_other_user_data(db_session)
-
+def test_token_endpoint_with_default_idp(client, persisted_refresh_tokens, auth_header):
     res = client.get("/token/?idp=default", headers=auth_header)
     assert res.status_code == 403
 
 
-def test_token_endpoint_with_idp_a(client, test_user, db_session, auth_header):
-    logged_in_user_data = create_logged_in_user_data(test_user, db_session)
-    create_other_user_data(db_session)
-
+def test_token_endpoint_with_idp_a(client, persisted_refresh_tokens, auth_header):
     # the token returned for a specific IDP should be created using the
     # corresponding refresh_token, using the logged in user's username
     res = client.get("/token/?idp=idp_a", headers=auth_header)
     assert res.status_code == 200
-    assert (
-        res.json["token"]
-        == "access_token_for_" + logged_in_user_data["idp_a"]["refresh_token"]
-    )
+
+    original_refresh_token = persisted_refresh_tokens["idp_a"][0]["refresh_token"]
+    assert res.json["token"] == f"access_token_for_{original_refresh_token}"
 
 
 def test_token_endpoint_without_specifying_idp(
-    client, test_user, db_session, auth_header
+    client, persisted_refresh_tokens, auth_header
 ):
-    logged_in_user_data = create_logged_in_user_data(test_user, db_session)
-    create_other_user_data(db_session)
-
     res = client.get("/token/", headers=auth_header)
     assert res.status_code == 403
 
 
-def test_token_endpoint_with_bogus_idp(client, test_user, db_session, auth_header):
-    logged_in_user_data = create_logged_in_user_data(test_user, db_session)
-    create_other_user_data(db_session)
-
-    # make sure the IDP we use is "default" when no IDP is requested
+def test_token_endpoint_with_bogus_idp(client, persisted_refresh_tokens, auth_header):
     res = client.get("/token/?idp=bogus", headers=auth_header)
     assert res.status_code == 400
 
 
-def test_token_endpoint_without_auth_header(client, test_user, db_session):
-    logged_in_user_data = create_logged_in_user_data(test_user, db_session)
-    create_other_user_data(db_session)
-
+def test_token_endpoint_without_auth_header(client, persisted_refresh_tokens):
     res = client.get("/token/")
     assert res.status_code == 403
 
 
+def assert_authz_mapping_for_test_user_in_default_commons(authz_mapping):
+    assert "/1" in authz_mapping
+    assert "/2" not in authz_mapping
+    assert "/3" not in authz_mapping
+    assert "/4" not in authz_mapping
+    assert "/5" not in authz_mapping
+
+
+def assert_authz_mapping_for_test_user_in_idp_a_commons(authz_mapping):
+    assert "/3" in authz_mapping
+    assert "/1" not in authz_mapping
+    assert "/2" not in authz_mapping
+    assert "/4" not in authz_mapping
+    assert "/5" not in authz_mapping
+
+
+def test_aggregate_user_user_endpoint(
```

Comment:
i know all the tests have been in the same file until now but i think it'd be good to put the aggregate endpoint tests in their own file